csharp/src/Ice/Communicator.cs

```diff
@@ -115,7 +115,7 @@ public bool isShutdown()
     /// <returns>The proxy.</returns>
     public ObjectPrx? propertyToProxy(string property)
     {
-        string proxy = instance.initializationData().properties.getProperty(property);
+        string proxy = instance.initializationData().properties!.getProperty(property);
```

Comment:
All these "!" derive from the design decision to hold onto an InitializationData in Instance, as opposed to a different data structure.  There is no alternative unless we refactor all this code.